### PR TITLE
CStoppableSGObject class

### PR DESCRIPTION
**Pull request by shubham808**
[WIP] This simply takes all premature stopping code and places it in a new class. Most of the python legacy examples broke locally. I am struggling with the swig part.

### CODE REVIEW

File: src/shogun/machine/Machine.cpp [hunk 1]
```diff
@@ -14,9 +14,9 @@
 using namespace shogun;
 
 CMachine::CMachine()
-    : CSGObject(), m_max_train_time(0), m_labels(NULL), m_solver_type(ST_AUTO),
-      m_cancel_computation(false), m_pause_computation_flag(false)
+    : CStoppableSGObject(), m_max_train_time(0), m_labels(NULL), m_solver_type(ST_AUTO)
 {
+  init_stoppable();
```
Comment (vigsterkr) [inline src/shogun/machine/Machine.cpp]: tab?

**Comment by shubham808** [comment]
@vigsterkr see CMachineEvaluation and CMachine. There is some duplicate code. (also CMachineEvaluation can be a non machine use case too)

**Comment by geektoni** [comment]
Great! :+1: I've added some comments, but it seems okay already. I guess that for fix the errors on Travis you should add this new file "StoppableSGObject.h" to SWIG (since here [https://travis-ci.org/shogun-toolbox/shogun/jobs/377448506#L1715](https://travis-ci.org/shogun-toolbox/shogun/jobs/377448506#L1715) CStoppableSGObject is ignored). Please have a look at [SGBase.i](https://github.com/shogun-toolbox/shogun/blob/develop/src/interfaces/swig/SGBase.i) to see how it's done for other base classes

File: src/shogun/machine/Machine.h [hunk 2]
```diff
@@ -147,7 +142,7 @@ enum EProblemType
  * locking.
  *
  */
-class CMachine : public CSGObject
+class CMachine : public CStoppableSGObject
```
Comment (karlnapf) [inline src/shogun/machine/Machine.h]: @vigsterkr @lisitsyn can we start using multiple inheritance in the c++ framework? Or is there still a blocker?

**Comment by karlnapf** [comment]
Nice one!

**Pull request merged**